presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PullUpExpressionInLambdaRules.java

```diff
@@ -295,12 +296,13 @@ private RowExpression getArgumentForRegexTypeExpression(RowExpression rowExpress
         @Override
         public Boolean visitSpecialForm(SpecialFormExpression specialForm, Boolean context)
         {
-            if (UNSUPPORTED_TYPES.contains(specialForm.getForm())) {
+            if (UNSUPPORTED_TYPES.contains(specialForm.getForm()) || specialForm.getArguments().size() > MAX_NUM_OF_ARGUMENTS) {
```

Comment:
There are many other visitors that seem to work fine with this kind of expression. So I suggest looking closer into what you are doing here that's causing the slowdown. Maybe some inefficient data structures? Let's root cause this.

Reply:
Root cause that the main contribution of the latency is the List contains check commented below. Fix this part to use a Set instead.

Reply:
@feilong-liu is the limit still needed

Reply:
Removed, I do not think it's needed now.